### ISSUE TITLE
Sickrage daily search not working, backlog search does

### ISSUE THREAD

**Issue by vinyli** [SiCKRAGETV/sickrage-issues#3559]
Branch: master
Commit hash: (7ce234c22da08a9912e968c38794a85584a83ce2)
Your operating system and python version: QTS 4.2.0 and Python 2.7 (r27:82500, Aug 21 2010, 10:10:43) [GCC 4.2.3]

My daily search isn't working. I've searched everywhere on the internet, but can't find a solution for the problem. I also put it on the forum, but no one is answering it there. I did some researching myself and i think i know what the problem is, but can't solve it. What i found out is that that the cache update link always adds/contains the category 5060 (Sports) and 5070 (Anime), even when you don't select the categories in the Search Provider section. In this case, the XML is relatively short and only contains Sports. When you manually delete the category 5060 in the link, the XML is much longer and contains everything i need. Sadly, i can't figure out how to delete that category 5060 from the cache update link.
In short: When i only add the category 5000 (TV) to my search provider my cache update URL looks like this: http://192.168.1.107/spotweb/api?apikey=**********&maxage=4&t=tvsearch&cat=5040%2C5060%2C5070 and the xml only contains sports and my log gets filled with 'Unable to parse filename' rules.

In a backlog search the URL doesn't add the %2C5060%2C5070 part and it snatches flawlessly.

I hope someone can help me out, i've been spending hours to get it right, but can't seem to nail it. :-(

Here is a log of a daily search:
https://gist.github.com/vinyli/9e48dc30d6ddc6b3325a

**Comment by SiCKRAGEB0T** [comment]
## Thanks for the bug/issue report
Please make sure to submit all of the [required information](https://github.com/SiCKRAGETV/sickrage-issues/blob/master/README.md), or we won't be able to help you!
Provide as **much** information as you can such as debug logs, commit hash, and tell us how to replicate the problem.For basic **questions** use the [ [forum] ](https://sickrage.tv/forums/), and for **Feature Requests** [ [Feathub] ](http://feathub.com/SiCKRAGETV/SickRage)
Also make sure the title accurately describes your issue. ("SR not working" etc. is not ok!)

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [x] Post debug logs, either inline (for smaller logs) or using gist)

## Many issues can simply be solved by:
- Making sure you update to the latest version.
- Have you tried turning your device off and on again?
- Using the **search** function to see if this issue has already been reported/solved.
- Do a [GIT reset](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#update-problems-try-this).
- Checking the [ [Wiki] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki) for [ [Guides] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Installation-&-Configuration-Guides), [ [FAQS] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes) [ [Main settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Settings-explained), [ [Show settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Show-settings-explained) [ [Remaining settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Remaining-settings-explained)
- Shutting down SickRage and removing the `cache/mako` & `cache/sessions` folders.
- If SickRage doesnt find any episodes you probably need to :
 - add a [Scene Exception](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Scene-exceptions-and-numbering)
 - Set the [Default Episode Status](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#newly-aired-shows-are-not-downloading-and-set-to-skippedignored) to `WANTED`
 - Make sure you understand how [quality detection](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Quality-Settings#quality-detectiondetermination) works during searches.
 - Shutdown SickRage and remove the `cache.db` file.
- If you use a QNAP and have SSL errors try this [package](http://apps.qnap.community/11-community/17-qsickrage).
- If you have a `'ascii' codec can't encode character` error, than set your Locale to [UTF-8](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#i-have-problems-with-special-characters-%C3%A9-etc-what-can-i-do)

*ps. If you enjoy Sickrage and like the support provided, than please consider making a [donation](https://github.com/SiCKRAGETV/SickRage/wiki/Donations) to keep SickRage alive.*

**Comment by neoatomic** [comment]
@vinyli Indeed the category 5060 (Sports) and 5070 (Anime) get added automatically to the one you yourself add.
This problem is only existing with spotweb.? newznab servers are working ok.?

@miigotu I'm also not happy that those 2 get added by default, can't we make it so that when sports or anime is enabled that those get added to the search string.? Its more efficient and reduces processing time.

**Comment by vinyli** [comment]
@neoatomic thanks for your quick response. Ok, but can you explain how it is possible that if the category 5060 is added in the link i only get sports (and nothing else) in my XML and when i manually delete the 5060 in in the browser my xml is way bigger and really contains all the categories i ask for (and the shows i want to snatch)?

I tested it with OZnzb (the only other newznab i can test) and that seems to be fine (the xml contains more than only Sports), so the problem only seems to be with Spotweb as far as i can see. I have my own Spotweb on my Nas, so it would be nice if this could work.

**Comment by neoatomic** [comment]
@vinyli You answer your own question. :P 
If you change the URL than it works. Meaning the course is spotweb not SickRage. 
And we have no control over what your spotweb installation returns in its url.
Maybe its a setting or limitation in spotweb that limits the results etc. Cant tell as i dont use spotweb myself...

Lets wait to see if miigotu sees any options here.

**Comment by vinyli** [comment]
@neoatomic it's not just my Spotweb installation. This problem is with other Spotwebs i have as Search Providers too, so it's not just my install that is the problem. I use mostly Spotweb because it contains nearly all the releases of tv shows with dutch subs (i'm from the Netherlands). Other newznab sites i know have less, so it would be nice if this would work.

But as you say, we'll wait what miigotu says. I hope he is willing to implement your option above, leave 5060 and 5070 out. Thank you for your input! :-)

**Comment by miigotu** [comment]
Ok so here is the problem:
There is no 5070 (anime) in the github version:
https://github.com/spotweb/spotweb/blob/d99e2fb2707409f60821d9b9df8b5d3345d6a3eb/lib/page/SpotPage_newznabapi.php#L824

Also, the site we use to test spotweb, is spotweb.be, who returns results either way, but does not filter results properly (you can search for season=1&ep=1 and it just returns everything. In fact the regex they use on the ep= value is even wrong and requires an e or E like `e1`, and it still fails even)

Either fix the newznab support for spotweb, or dont complain that it doesnt work as expected =P

**Comment by vinyli** [comment]
@miigotu i'm sorry but what do you mean by 'fix the newznab support for spotweb'? Is it something i can do myself? Or is this a Spotweb problem in general? It's not meant as a complaint. I'm just wondering why it doesn't work. :-)

**Comment by neoatomic** [comment]
@vinyli Spotweb doesn't have a 5070 (anime) category in there code. So one needs to be created in there code. ( And i just note they also dont have a web-dl category :( )

We zijn aan het kijken wat mogelijk is. :)
I see miigotu coednt resist and pushed a fix for another issue. https://github.com/spotweb/spotweb/pull/102/files

**Comment by miigotu** [comment]
I have anime added also, as well as a fix for wrong content coming back in selected categories. Still finishing that one up.

**Comment by miigotu** [comment]
https://github.com/spotweb/spotweb/pull/103/files

**Comment by vinyli** [comment]
Ok, if i understand it correctly, there is nothing i can do to make the Daily Search work with Spotweb?

@neoatomic Das mooi, bedankt voor je hulp ;-)

**Comment by neoatomic** [comment]
Daily Search works fine, Spotweb however doesn't return results for category 5070.
If you check the link above you will see that miigotu was so kind to submit a fix/commit for the Spotweb developers to create that 5070 category. They only need to approve/add those changes to the Spotweb code. 
When that is done you need to update spotweb to the newest version. Thats it.

**Comment by vinyli** [comment]
I know, @miigotu thanks for making that issue! And this program is really cool.
I appreciate you guys trying to help me out, after all you do it in your spare time (and if i get annoying please say so, then i'll leave it this way :-))!

But i get a feeling we don't quite understand each other (or i don't get it, because my knowledge isn't far as good as you guys about this, which could well be the case). My problem is with the 5060 category that is being added. Strange thing is that i did some other testing and when i add the category 5060 Sports to the search provider the url contains the category twice (the one i added in the search provider and the one that is added automatically). In this case, my XML only contains sports, no matter what extra categories are defined in the Search provider. If i delete the 5060 Sports category manually in the url in the webbrowser that you guys add, the xml contains everything (even when the first 5060 Sports added in the Search provider stays in the url) so i don't understand what this has to do with category 5070? 

Or is my whole problem existing because this category 5070 Anime is not in Spotweb?
Also, i almost can't believe there is no one else with this problem (because no one is reacting here besides you guys. I guess everyone who uses Spotweb as a Newznab should have this problem.

**Comment by miigotu** [comment]
Thats another part of the issue. You dont need to add the sports category to your category list, it is tacked on automatically when searching for a show marked as sports.

**Comment by vinyli** [comment]
Hey @miigotu, i implemented your solution in my Spotweb installation and in my XML i now only get Sports and Anime (only the categories that are automatically added), nothing else (the problem also occurs on Spotweb.be, but them in it's original form with the Sports, so it's not just my installation). 

I'v added the xml here:

https://gist.github.com/vinyli/29d2675a466b4bea93e6

Also, i did some changing in my SpotPage_newznabapi.php and changed the categories behind '5060' (with no 5070 category in it) in this:

case 5060: return 'cat0_z1,cat0_z1_a0,cat0_z1_a1,cat0_z1_a2,cat0_z1_a3,cat0_z1_a10,~cat0_z1_d18';

and then the xml only shows all these categories. So it seems it shows only the spots in the automatically added categories.
